chalk-ir/src/fold/boring_impls.rs

```diff
@@ -9,25 +9,25 @@ use crate::*;
 use std::marker::PhantomData;
 use std::sync::Arc;
 
-impl<'a, T: Fold<I, TI>, I: Interner, TI: TargetInterner<I>> Fold<I, TI> for &'a T {
+impl<'a, T: Fold<I, TI> + Clone, I: Interner, TI: TargetInterner<I>> Fold<I, TI> for &'a T {
```

Comment:
Where is this impl actually used?

Reply:
This might be hiding a lot of cloning.

Reply:
It does, which is why I was thinking about removing it. I'm not sure where it's used, but replacing the code with `unreachable!()` makes most of the tests fail.